syntax = "proto3";

package nodeservice;

service NodeService{
    rpc JoinNetwork (Address) returns (ResponseMessage){}
    rpc LeaveNetwork(DefaultRequest) returns (ResponseMessage){}
    rpc GetFingerTable(DefaultRequest) returns (ResponseMessage){}
    rpc GetPredSucc(DefaultRequest) returns (ResponseMessage){}
    rpc LookUpID(NodeId) returns (JoinNodeResponse){}
    rpc ConnectPeer(Address) returns (Address){}
    rpc UpdateFingerTable(DefaultRequest) returns(Address){}
    rpc UpdatePredSucc(UpdatePredSuccRequest) returns(Address){}
    rpc SearchFile (SearchFileRequest) returns (SearchFileResponse) {}
}


message DefaultRequest{}
message NodeId{
    int32 id = 1; 
}

message ResponseMessage{
    string message = 1;
}

message UpdatePredSuccRequest{
    int32 identifier = 1;
    Address address = 2;
}

message JoinNodeResponse{
    int32 identifier = 1;
    Address address = 2;
}

message Address{
    string ip = 1;
    int32 port = 2;
}

// Nuevos mensajes para b√∫squeda de archivos
message SearchFileRequest {
  string filename = 1;
}

message SearchFileResponse {
  repeated FileInfo files = 1;
}

message FileInfo {
  string name = 1;
  string uri = 2;
}

message UploadRequest {
  string filename = 1;
}

message UploadResponse {
  string message = 1;
}